bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/EcobeeBinding.java

```diff
@@ -0,0 +1,1023 @@
+			 * thermostat, these tests will be perfectly accurate.
+			 */
+
+			boolean changed = false;
+
+			changed = changed
+					|| (newRevision.hasRuntimeChanged(lastRevision) && (selection.includeRuntime() || selection
+							.includeExtendedRuntime()));
+			changed = changed
+					|| (newRevision.hasThermostatChanged(lastRevision) && (selection.includeSettings() || selection
+							.includeProgram()));
+
+			if (changed) {
+				thermostatIdentifiers.add(newRevision.getThermostatIdentifier());
+			}
+		}
+
+		// Remember the new revisions for the next execute() call.
+		this.lastRevisionMap = newRevisionMap;
+
+		if (0 == thermostatIdentifiers.size()) {
+			logger.debug("No changes detected.");
+		} else {
+			logger.debug("Requesting full retrieval for {} thermostat(s).", thermostatIdentifiers.size());
+
+			// Potentially decrease the number of thermostats for the full
+			// retrieval.
+
+			selection.setSelectionMatch(thermostatIdentifiers);
+
+			// TODO loop through possibly multiple pages
+			ThermostatRequest treq = new ThermostatRequest(oauthCredentials.accessToken, selection, null);
+			ThermostatResponse tres = treq.execute();
+
+			if (tres.isError()) {
+				logger.error("Error retrieving thermostats: {}", tres.getStatus());
+			} else {
+
+				// Create a ID-based map of the thermostats we retrieved.
+				Map<String, Thermostat> thermostats = new HashMap<String, Thermostat>();
+
+				for (Thermostat t : tres.getThermostatList()) {
+					thermostats.put(t.getIdentifier(), t);
+				}
+
+				// Iterate through bindings and update all inbound values.
+				for (final EcobeeBindingProvider provider : this.providers) {
+					for (final String itemName : provider.getItemNames()) {
+						if (provider.isInBound(itemName)) {
+							final State newState = getState(provider, thermostats, itemName);
+							ItemRegistry reg = provider.getItemRegistry();
+							State oldState = reg.getItem(itemName).getState();
+
+							if ((oldState == null && newState != null)
+									|| (UnDefType.UNDEF.equals(oldState) && !UnDefType.UNDEF.equals(newState))
+									|| !oldState.equals(newState)) {
+								logger.debug("readEcobee: Updating itemName '{}' with newState '{}', oldState '{}'",
+										itemName, newState, oldState);
+
+								/*
+								 * we need to make sure that we won't send out this event to Ecobee again, when
+								 * receiving it on the openHAB bus
+								 */
+								ignoreEventSet.add(itemName + newState.toString());
+								logger.trace("Added event (item='{}', newState='{}') to the ignore event list",
+										itemName, newState);
+								this.eventPublisher.postUpdate(itemName, newState);
+							} else {
+								logger.trace("readEcobee: Ignoring item='{}' with newState='{}', oldState='{}'",
+										itemName, newState, oldState);
+							}
+						}
+					}
+				}
+			} // end if there was no error retrieving the thermostats
+		} // end if there were any changed thermostats to fetch
```

Comment:
please remove these comments. Either refactor the code in order to reduce indention depth or stand it like a man ;-) (i would prefer option 1)